.github/workflows/publish.yml

```diff
@@ -1,22 +1,66 @@
 on:
   push:
-    tags:
-      - "*"
+    branches:
+      - main
 
 # These permissions are needed to assume roles from Github's OIDC.
 permissions:
   contents: write
   id-token: write
+  pull-requests: write
 
 name: Publish Extension
 jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created: "${{ steps.release-please.outputs.release_created }}"
+      release_tag: "${{ steps.release-please.outputs.tag_name }}"
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+        with:
+          # https://github.com/actions/checkout/issues/1467
+          fetch-depth: 0
+
+      - uses: googleapis/release-please-action@7987652d64b4581673a76e33ad5e98e3dd56832f # v4.1.3
+        id: release-please
+        with:
+          release-type: node
+          default-branch: main
+          pull-request-title-pattern: "Automatic release v${{ version }} 🚀"
+          github-token: ${{ github.secret }}
+          bump-minor-pre-major: true # Avoids to generate a major version. Remove when necessary
+          include-v-in-tag: true
+          draft-pull-request: true
+          changelog-types:
+            feat: "🎉 Features"
+            fix: "🐛 Bug Fixes"
+            deps: "🔗 Dependencies"
+            docs: "📝 Documentation"
+            ci: "🤖 Continuous Integration"
+            refactor: "♻️ Code Refactoring"
+            test: "✅ Tests"
+            revert: "⏪ Reverts"
+            style: "💄 Style"
+            build: "🏗️ Build System"
+            perf: "⚡ Performance Improvements"
+            chore: "🔧 Miscellaneous Chores"
+  
   lint:
+    needs: release-please
+    if: needs.release-please.outputs.release_created
```

Comment:
Not sure why this job should only be run after `release-please` and if a release was created.

I think it should either run for anything pushed to main, or be deleted (since we already run linters and a compilation check in the CI pipeline for PRs)

Reply:
This one was executed for every tag, but I'm seeing that its the same like: https://github.com/grafana/grafana-vs-code-extension/blob/main/.github/workflows/ci.yml.

So maybe we could remove it from here and add main branch too in the other yaml?